### PR TITLE
Ensure that the TensorPrimitives.Min/MaxNumber functions follow the IEEE 754:2019 spec

### LINKED ISSUE #102156: Test failure for TensorPrimitives MinNumber and MaxNumber on Arm64
## Build Information
Build: https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_build/results?buildId=673236
Build error leg or test failing: System.Numerics.Tensors.Tests.SingleGenericTensorPrimitives.SpanSpanDestination_SpecialValues
Pull request: https://github.com/dotnet/runtime/pull/102098

## Error Message

Fill the error message using [step by step known issues guidance](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssueJsonStepByStep.md).

```json
{
  "ErrorMessage": "",
  "ErrorPattern": "Single (MaxNumber|MinNumber)\\(Single, Single\\).*\\[FAIL\\]",
  "BuildRetry": false,
  "ExcludeConsoleLog": false
}
```

 ### Known issue validation
**Build: :mag_right:** https://dev.azure.com/dnceng-public/public/_build/results?buildId=673236
**Error message validated:** `[Single (MaxNumber|MinNumber)\(Single, Single\).*\[FAIL\]`]
**Result validation:** :white_check_mark: Known issue matched with the provided build.
**Validation performed at:** 5/13/2024 1:17:52 PM UTC

### Report

|Build|Definition|Test|Pull Request|
|---|---|---|---|
|[673236](https://dev.azure.com/dnceng-public/public/_build/results?buildId=673236)|dotnet/runtime|[System.Numerics.Tensors.Tests.SingleGenericTensorPrimitives.SpanSpanDestination_SpecialValues](https://dev.azure.com/dnceng-public/public/_build/results?buildId=673236&view=ms.vss-test-web.build-test-results-tab&runId=16696422&resultId=183164)|dotnet/runtime#102098|
|[2449946](https://dev.azure.com/dnceng/internal/_build/results?buildId=2449946)|dotnet-runtime|[System.Numerics.Tensors.Tests.SingleGenericTensorPrimitives.SpanSpanDestination_SpecialValues](https://dev.azure.com/dnceng//internal/_build/results?buildId=2449946&view=ms.vss-test-web.build-test-results-tab&runId=53594597&resultId=161191)||
#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|2|2|2|

### PR DESCRIPTION
**Pull request by tannergooding**
This resolves #102156

### CODE REVIEW

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.MaxNumber.cs [hunk 1]
```diff
@@ -87,20 +88,30 @@ public static Vector128<T> Invoke(Vector128<T> x, Vector128<T> y)
             {
                 if (typeof(T) == typeof(float) || typeof(T) == typeof(double))
                 {
-                    if (AdvSimd.IsSupported && typeof(T) == typeof(float))
+                    // We can't use AdvSimd.MaxNumber here because it doesn't correctly
+                    // handle sNaN (it converts it to qNaN as per the now deprecated
+                    // maxNum function defined by IEEE 754:2008, but which is not inline
+                    // with the maximumNumber function that replaces it in IEEE 754:2019
```
Comment (stephentoub) [inline src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.MaxNumber.cs:94]: Nit: missing closing paren

**Review by stephentoub — APPROVED** [review]
Thanks!

**Review by eiriktsarpalis — APPROVED** [review]

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.MaxNumber.cs [hunk 2]
```diff
@@ -87,20 +88,30 @@ public static Vector128<T> Invoke(Vector128<T> x, Vector128<T> y)
...
+
+                    Vector128<T> max;
+
+                    if (Sse.IsSupported && typeof(T) == typeof(float))
```
Comment (eiriktsarpalis) [inline src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.MaxNumber.cs:98]: Nit: presumably this change is unrelated to the test fix at hand, would it be possible to update the PR title/commit message to include this change?

Comment (tannergooding) [reply src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.MaxNumber.cs:98]: It's not a change. It's preserving the behavior and perf for xarch from prior to this PR

Comment (tannergooding) [reply src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.MaxNumber.cs:98]: That is, `Vector128.Min` on xarch currently emits `minps`/`minpd` (`Sse.Min` for float and `Sse2.Min` for double) while on Arm64 it emits `fmin`.

We therefore cannot use `Vector128.Min` on Arm64 since it propagates NaN (and longterm wouldn't work for xarch either, as we're working towards ensuring consistency there). But, it's safe to continue using the underlying intrsinsics for xarch and thus allows us to preserve codegen as-is there.
